internal/watcher/watcher.go

```diff
@@ -169,15 +196,61 @@ func (d Differential) sendDiff(path string, diff string) error {
 	return nil
 }
 
+// WatchFn for reading files in the watched folder and look for difference from the last update
 func (d Differential) WatchFn(path string) error {
 	if _, ok := d.ignoredExtensions[filepath.Ext(path)]; ok {
 		return nil
 	}
 
-	// This is to allow for changes to propagate on the filesystem. If the file
-	// is large, the write won't be atomic. It will happen in, for example, 4096
-	// bytes chunks. 1 ms should be enough.
-	time.Sleep(time.Millisecond)
+	time.Sleep(10 * time.Millisecond)
+
+	info, err := os.Stat(path)
+	if err != nil {
+		return fmt.Errorf("error read file status %w", err)
+	}
+
+	if info.Size() >= MaxFileSize {
```

Comment:
Should these two code paths be unified? If there are a lot of small files, the server can still run out of memory. This would unify the use of `previousContents` which can be renamed to `previousContentPath` then.